src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -156,19 +168,54 @@ public Path exportProgrammingExerciseInstructorMaterial(ProgrammingExercise exer
     }
 
     /**
-     * Export instructor repositories and optionally students' repositories in a zip file.
+     * In case the problem statement contains embedded files, they need to be part of the zip, so they can be imported again.
      *
+     * @param exercise        the programming exercise that is exported
+     * @param outputDir       the directory where the content of the export is stored
+     * @param pathsToBeZipped the paths that should be included in the zip file
+     */
+
+    private void copyEmbeddedFiles(ProgrammingExercise exercise, Path outputDir, List<Path> pathsToBeZipped) throws IOException {
+        List<String> embeddedFiles = new ArrayList<>();
```

Comment:
This should be a set in case there is the same file in the problem statement twice